File: HotelAPI/src/main/java/br/com/totvs/hotel/model/cliente/ClienteModel.java
package br.com.totvs.hotel.model.cliente;

import br.com.totvs.hotel.model.estadia.EstadiaModel;
import br.com.totvs.hotel.model.pessoa.PessoaModel;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "table_cliente")
public class ClienteModel extends PessoaModel {
    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String celular;

    @OneToMany(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_cliente")
    private List<EstadiaModel> estadias = new ArrayList<EstadiaModel>();

}

File: Hotel-Interface/accor/src/app/model/cliente.model.ts
import { Endereco } from "./endereco.model";

export class Cliente {
    id: string;
    nome: string;
    sobrenome: string;
    rg: string;
    cpf: string;
    dataNascimento: string;
    endereco: Endereco;
    email: string;
    celular: string;

    constructor(id: string, nome: string, sobrenome: string, rg: string, cpf: string, dataNascimento: string, endereco: Endereco, email: string, celular: string) {
        this.id = id;
        this.nome = nome;
        this.sobrenome = sobrenome;
        this.rg = rg;
        this.cpf = cpf;
        this.dataNascimento = dataNascimento;
        this.endereco = endereco;
        this.email = email;
        this.celular = celular;
    }
}
